source/common/config/utility.cc

```diff
@@ -11,18 +11,27 @@
 namespace Envoy {
 namespace Config {
 
-namespace {
-
-void translateApiConfigSource(const std::string& cluster, uint32_t refresh_delay_ms,
-                              envoy::api::v2::ApiConfigSource& api_config_source) {
-  api_config_source.set_api_type(envoy::api::v2::ApiConfigSource::REST_LEGACY);
+const std::string Utility::REST_LEGACY = "REST_LEGACY";
+const std::string Utility::REST = "REST";
+const std::string Utility::GRPC = "GRPC";
+
+void Utility::translateApiConfigSource(const std::string& cluster, uint32_t refresh_delay_ms,
+                                       const std::string& api_type,
+                                       envoy::api::v2::ApiConfigSource& api_config_source) {
+  // TODO(junr03): document the option to chose an api type once we have created
+  // stronger constraints around v2.
+  if (api_type == REST_LEGACY) {
+    api_config_source.set_api_type(envoy::api::v2::ApiConfigSource::REST_LEGACY);
+  } else if (api_type == REST) {
+    api_config_source.set_api_type(envoy::api::v2::ApiConfigSource::REST);
+  } else if (api_type == GRPC) {
```

Comment:
This should probably be:

```
} else {
  ASERT(api_type == GRPC);
  ...
}
```

since the JSON schema knows this is an enum and has validated.